src/app/content/highlights/utils/summaryHighlightsUtils.ts

```diff
@@ -38,12 +39,23 @@ interface DataRemove extends BaseData {
   id: string;
 }
 
-export const removeSummaryHighlight = (summaryHighlights: SummaryHighlights, data: DataRemove) => {
+export const removeSummaryHighlight = (
+  summaryHighlights: SummaryHighlights,
+  data: DataRemove
+): [SummaryHighlights, Highlight | null] => {
   const { locationFilterId, pageId, id } = data;
 
   const filteredHighlights = summaryHighlights[locationFilterId] && summaryHighlights[locationFilterId][pageId]
     ? summaryHighlights[locationFilterId][pageId].filter((highlight) => highlight.id !== id)
-    : [];
+    : null;
+
+  const removedHighlight = summaryHighlights[locationFilterId] && summaryHighlights[locationFilterId][pageId]
+    ? summaryHighlights[locationFilterId][pageId].find((highlight) => highlight.id === id)
+    : null;
+
+  if (!filteredHighlights || !removedHighlight) {
+    return [summaryHighlights, null];
+  }
```

Comment:
This could be done in the same iteration:
```
const highlights = summaryHighlights[locationFilterId] && summaryHighlights[locationFilterId][pageId];
  if (!highlights) { return [summaryHighlights, null]; }

  const filteredHighlights: Highlight[] = [];
  let removedHighlight: Highlight | null = null;

  for (const highlight of highlights) {
    if (highlight.id !== id) { filteredHighlights.push(highlight); }
    removedHighlight = highlight;
  }

  if (!removedHighlight) { return [summaryHighlights, null]; }
```